test_runner/regress/test_layer_writers_fail.py

```diff
@@ -29,17 +30,55 @@ def test_image_layer_writer_fail_before_finish(neon_simple_env: NeonEnv):
         ]
     )
 
-    wait_for_last_flush_lsn(env, pg, tenant_id, timeline_id)
-    pageserver_http.timeline_checkpoint(tenant_id, timeline_id)
+    pageserver_http.configure_failpoints(("image-layer-writer-fail-before-finish", "return"))
+    with pytest.raises(Exception, match="image-layer-writer-fail-before-finish"):
+        pageserver_http.timeline_checkpoint(tenant_id, timeline_id)
 
-    old_layer_files = [path for path in env.timeline_dir(tenant_id, timeline_id).iterdir()]
+    is_temp_path = lambda file: str(file)[-7:] == '___temp'
```

Comment:
This particular prefix is pageserver's knowledge: https://github.com/neondatabase/neon/blob/259a5f356e036b5cf274cd2222884434dd8c55f0/pageserver/src/lib.rs#L128 so we might want to put it as a const in `neon_fixtures.py::NeonPageserver`.

One idea for future would be to check every timeline and tenant creation failure test for absence of temp files, that constant would help to do it later.